clippy_lints/src/declared_lints.rs

```diff
@@ -546,6 +546,7 @@ pub(crate) static LINTS: &[&crate::LintInfo] = &[
     crate::self_named_constructors::SELF_NAMED_CONSTRUCTORS_INFO,
     crate::semicolon_block::SEMICOLON_INSIDE_BLOCK_INFO,
     crate::semicolon_block::SEMICOLON_OUTSIDE_BLOCK_INFO,
+    crate::semicolon_block::SEMICOLON_OUTSIDE_BLOCK_IF_SINGLELINE_INFO,
```

Comment:
I feel like this shouldn't be a new lint, but much rather a configuration for `semicolon_outside_block`. When I see `#[warn(clippy::semicolon_outside_block_if_singleline)]` I would expect it to only lint single line cases, and not also multiline ones.


Now with a configuration, it would look something like this:

```
// main.rs
#![warn(clippy::semicolon_inside_block)]

// clippy.toml
semicolon-outside-block-for-single-line = true
```

Which is more effort, but that should be fine with a `restriction` lint. This way, it's clear that semicolons are usually preferred inside the block, unless it's a single line. The config name might need some improvement, but that would feel a bit better to me. What do you think? Also, because you suggested the lint in the first case? :)

Here is some documentation, if you want to implement a configuration value instead: [Adding configuration to a lint](https://doc.rust-lang.org/clippy/development/adding_lints.html#adding-configuration-to-a-lint)

Reply:
I've added that plus another for semicolon_inside_block, for consistency. with both of them enabled it's the same behavior as the original lint.